sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -364,6 +364,27 @@ static size_t _handle_req(coap_pkt_t *pdu, uint8_t *buf, size_t len,
             return gcoap_response(pdu, buf, len, COAP_CODE_INTERNAL_SERVER_ERROR);
             break;
     }
+    
+    if (resource->flags & COAP_SEPARATE_RESPONSE) {
+        coap_pkt_t pdu_sep;
+        uint8_t buf_sep[CONFIG_GCOAP_PDU_BUF_SIZE] = { 0 };
+        memcpy(buf_sep, buf, len);
+        ssize_t res = coap_parse(&pdu_sep, buf_sep, len);
+        if (res < 0) {
+            DEBUG("gcoap: parse failure: %d\n", (int)res);
```

Comment:
There is a tighter bound than CONFIG_GCOAP_PDU_BUF_SIZE on this (header plus maximal token).